bin/dry-run.rb

```diff
@@ -301,16 +301,18 @@ def cached_dependency_files_read
   commit: $options[:commit]
 )
 
-$repo_contents_path = nil
+$repo_contents_path = Dir.mktmpdir if $options[:clone]
```

Comment:
> is that better?

It's a little meh, because it also means that we won't fetch different branches or pull in updates if `origin main` has changed 🤔 I'm OK with the temp file every run